core/src/worker/mod.rs

```diff
@@ -937,6 +970,33 @@ impl Worker {
             self.wfts_drained_notify.notified().await;
         }
     }
+
+    /// Attempt to reserve activity slots for activities we could eagerly execute on
+    /// this worker.
+    ///
+    /// Returns the number of activity slots that were reserved
+    fn reserve_activity_slots_for_outgoing_commands(&self, commands: &mut [Command]) -> usize {
+        let mut reserved = 0;
+        for cmd in commands {
+            if let Some(Attributes::ScheduleActivityTaskCommandAttributes(attrs)) =
```

Comment:
@bergundy's discussion off-PR made me realize we need to probably fail if `ScheduleActivityTaskCommandAttributes` has `request_eager_execution` but has a different task queue than this one.